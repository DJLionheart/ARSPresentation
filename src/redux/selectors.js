import * as actions from './actions'
import paths from './paths'
import { getDispatch } from './store'

export const selectLuke = (state) => {
    const luke = state.getIn(paths.luke)
    console.log('selector', luke)
    if (luke === undefined) {
        getDispatch()(actions.fetchLuke())
        return false
    }

    return luke.toJS()
}