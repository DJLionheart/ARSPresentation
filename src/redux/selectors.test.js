import * as selectors from './selectors'
import { fromJS } from 'immutable';

describe('selectors', () => {
    test('selectors.selectLuke return luke', () => {
        const state = fromJS({
            luke: {
                name: 'Luke Skywalker',
                from: 'Swapi'
            }
        })

        expect(selectors.selectLuke(state)).toMatchSnapshot()
    })
})