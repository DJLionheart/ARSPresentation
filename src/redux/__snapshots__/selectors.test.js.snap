// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors selectors.selectLuke return luke 1`] = `
Object {
  "from": "Swapi",
  "name": "Luke Skywalker",
}
`;
